app/src/main/java/com/github/se/icebreakrr/ui/profile/InboxProfileView.kt

```diff
@@ -169,10 +180,18 @@ fun acceptDeclineCode(
     navigationActions: NavigationActions,
     uid: String,
     accepted: Boolean,
-    fcm: String
+    fcm: String,
+    location: Pair<Double, Double>,
+    locationMessage: String
 ) {
-  meetingRequestViewModel.setMeetingResponse(fcm, "accepting/decline request", accepted)
+  meetingRequestViewModel.setMeetingResponse(
+      fcm, "accepting/decline request", accepted, location.toString())
   meetingRequestViewModel.sendMeetingResponse()
+  if (accepted) {
+    meetingRequestViewModel.confirmMeetingLocation(uid, Pair(locationMessage, location)) {
+      Log.e("MeetingRequestService", "error when confirmMeetingLocation : ${it.message}")
+    }
+  }
   meetingRequestViewModel.removeFromMeetingRequestInbox(uid)
   meetingRequestViewModel.updateInboxOfMessages {}
```

Comment:
There could be racing conditions between these two functions too. The inbox won't be updated after `updateInboxOfMessages` since you modify your profile in the backend, and before it has finished modifying it, you get your profile that you consider as updated

Reply:
ok